lib/middleware/overload_protections.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Middleware
+  class OverloadProtections
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      is_logged_in = Auth::DefaultCurrentUserProvider.find_v1_auth_cookie(env).present?
```

Comment:
Note that this is cache in the `env` so we only pay the cost of this once per request: 

https://github.com/discourse/discourse/blob/deafbcdad91833b92dd4f4fffeef3b0e3e7de7b6/lib/auth/default_current_user_provider.rb#L77